[Documentation] Created a readme file for the QC module

knoll-alex opened the pull request:
## Brief summary of changes
Created a brand new readme file for the QC module

#### Testing instructions (if applicable)

1. Ensure information and formatting is correct!

#### Links to related tickets (GitHub, Redmine, ...)

https://github.com/aces/Loris/projects/6

AlexandraLivadas committed:
[Test] unittests for Breadcrumb.class.inc (#4769)

Add unit test for breadcrumb class.

AlexandraLivadas committed:
[Test] unittests for BreadcrumbTrail.class.inc (#4840)

Add basic unit test class for BreadcrumbTrail.class.inc which includes one unit test for the __toString() method.

ridz1208 committed:
Merge remote-tracking branch 'aces/21.0-release' into PUSH_v21.0.0.RC1_INTO_MAJOR

driusan committed:
Merge pull request #4933 from ridz1208/PUSH_v21.0.0.RC1_INTO_MAJOR

PUSH v21.0.0.RC1 INTO MAJOR

driusan committed:
Merge pull request #4961 from aces/minor

Pull minor into major

johnsaigle committed:
[Tools] Fix Raisinbread exporter location (#4922)

Resolves #4878 by changing the export location to be built using the base URL instead of using a relative path.

johnsaigle committed:
[DQT] Fix crash when CouchDB values not set in project/config.xml (#4897)

If project/config.xml file does not contain entries for CouchDB values, LORIS crashes with a blank page.

This prints a simple error message and logs a more informative error instead of crashing.

xlecours committed:
[API] Checking for logged in users (#4971)

This add a call to hasAccess function in /projects to lomit the access to logged in users only.

kongtiaowang committed:
[Test] unittest for NDB_Config  (#4383)

Add a new unit test for NDB_Config.

cmadjar committed:
[Clean up] deletion of the full MINC header entry from parameter_file as not used anymore (#4925)

Until now, the whole header of a MINC file was dumped into a row of parameter_file, taking a lot of space in the database and the SQL dump created for that table, while not being used at all.

During the imaging meeting of July 4th, a decision has been made to no longer store that row to gain some space in the database, hence, removing it from the imaging pipeline code.

This is linked to aces/Loris-MRI#453 PR on the LORIS-MRI side.

Note: the API does not look specifically for that header but dumps all the values present in the parameter_file table so no changes were needed.

jesscall committed:
[Core: Instrument] Fix bug related to altered return type (#4975)

Changing logic in functions getSubprojectID and getVisitLabel to accommodate a change in return type that was made to function getSessionID in instrument class, such that it's now strictly an integer.

johnsaigle committed:
[README] Change v 20 to v 21 (#4959)

Update reference to version number in README.

ridz1208 committed:
Merge remote-tracking branch 'aces/minor' into PUSH_MINOR_INTO_MAJOR_20190725

johnsaigle committed:
[Coding Standards] Enforce PSR2 standards on files in src/ (#4335)

The src/ directory and its children are intended to meet the PSR2 coding standards. This PR lays the foundation for achieving this by:

- Adding a new coding standards XML file implementing PSR2 checks on only the src/ directory
- Adding a line in the test script to be added to the analysis process
- Fixing all existing problems

driusan committed:
Merge pull request #4986 from ridz1208/PUSH_MINOR_INTO_MAJOR_20190725

PUSH MINOR INTO MAJOR

johnsaigle committed:
[Config] Add settingEnabled() helper function (#4940)

Add a new helper function settingEnabled to determine whether a boolean-style config setting is enabled or not.

This is designed to add some syntactic sugar and hide some of the implementation details of our Config library from calling code.

The calling code only cares whether a setting is enabled. Developers shouldn't need to concern themselves with doing a strict string type comparison on config settings.

e.g. this code:

if ($config->getSetting('useEDC') === 'true') {
    ...
}

can now be written as

if ($config->settingEnabled('useEDC')) {
    ...
}

AlexandraLivadas committed:
[Test] Adding to unittests for Visit and VisitController (#4988)

Adding to unit tests for the Visit.class.inc and VisitController.class.inc libraries.

cmadjar committed:
[SQL] Removed mri_acquisition_dates table from the database and the LORIS code (#4962)

This removes the mri_acquisition_dates table from the database as this table does not necessarily store the correct acquisition date. Instead, replaced all the queries to look for the minimal date present in the parameter_file table which is more robust than what was stored in mri_acquisition_dates.

Note: the decision to remove this table was taken during an imaging meeting in 2017 and confirmed at an other imaging meeting in 2019.

See also aces/Loris-MRI#463

driusan committed:
Merge pull request #5005 from aces/21.0-release

PUSH of 21.0-release into BUGFIX again

AlexandraLivadas committed:
[Test] Add to unittests for Candidate.class.inc (#4936)

Adding more unit tests for Candidate.class.inc in the CandidateTest.php file.

cmadjar committed:
[RB dataset] Updated test_battery of raisin bread (#4774)

This modifies the test_battery table of the raisin bread dataset so the test plan of Start Next Stage can be used on the RB dataset when testing on test.loris.ca. (Before, the test battery was too harmonized and we could not test all possibilities mentioned in the test plan.)

AlexandraLivadas committed:
[Test] unittests for Utility.class.inc (#4861)

Add unit tests for Utility class.

AlexandraLivadas committed:
[Test] Adding to unittests for Settings.class.inc (#4987)

Adding to unit tests for the Settings.class.inc library. There was only one test to add for the getBaseURL() method.

This completes the basic test coverage of the Settings class.

AlexandraLivadas committed:
[Test] Adding to unittests for LorisForm.class.inc (#4916)

Add unit tests for LorisForm.

AlexandraLivadas committed:
[Test] unittests for User.class.inc (#4979)

Unit tests for the User.class.inc class.

AlexandraLivadas committed:
[Test] Adding to unittests for NDB_Config.class.inc (#5012)

driusan committed:
Fix make clean build target (#5011)

The vendor directory created by composer is a directory, not
a file, so it needs to be deleted with "rm -rf", not "rm -f".

driusan committed:
Show error when installdb.php can not connect to database (#5023)

The code which checks if the connection to the database failed
was assuming $db->connect() throws an Exception. However,
$db->connect() catches all exceptions and returns false.

Check return value so that an error is displayed to the user
if the credentials are incorrect on the first page of installer.

johnsaigle committed:
[Core] Create File Uploader class to standardize and validate file uploads in LORIS (#4181)

This PR picks up on #3514 and introduces an attempt at creating a generic File Upload (Validator) class for use in LORIS file uploads. See attached issue for more details.

The newly Reactified instrument_manager is used as an example to demonstrate how this class should be used.

ridz1208 committed:
[tools/raisinbread exporter] Use config.xml credentials (#4969)

The previous iteration of the script was dependent on the existence of a mysql configuration file (.my.cnf for example). this instead makes it depend on the quat values in the configuration file

jesscall committed:
[Examiner] Bugfix for pselectRow returning multiple rows (certification_history) (#4981)

Bugfix for DomainException that was being thrown in examiner module. Please see ticket that is linked below for full description of the issue that this resolves.

jesscall committed:
[Core/Instrument] added default date value to getCandidateAge function (#5007)

Since the documentation in Instrument class says that date is optional for the function getCandidateAge(), i think it's best to add a default date = null.

jesscall committed:
[Media Uploader] Bugfix date of administration field (#4999)

This PR fixes issue reported in #4998. When entering a date in the past and attempting to upload a file, you get an error saying that the date cannot be in the future.

date_diff is used to find the difference between the date entered and now.
$now - $date returns a positive number if $date is in the past, therefore the PR simply switches the comparison operator.

AlexandraLivadas committed:
[Documentation] Added markdown file for Unit Testing Guide (#5066)

Unit testing guide written as part of Google Summer of Code project.

zaliqarosli committed:
add month input selector (#4994)

In Firefox polyfill for datepicker, only apply the jQuery fallback for month selection, not date selection (which Firefox now natively supports.)

jesscall committed:
[candidate_parameter] Consent Status data validation bugfix (#5000)

This fixes the logic for data validation on consent status tab. As it stands, if no status stays no or record existed as NULL the validation checks that :

- consent date is not empty
- withdrawal date is empty

However, if Consent Status at one point changes from 'YES' to 'NO' ==> user must fill in date of withdrawal. Therefore, consents that = 'NO' and have non-empty dates of withdrawal are considered non-valid (Displays error message).

kongtiaowang committed:
[install] Install.sh - Unbound Variable errors in Bash scripts (#5159)

Fix error about unbound variable "projectname" in install script.

johnsaigle committed:
[Test] Add link to Raisinbread modification instructions to the CONTRIBUTING doc (#5025)

Add Raisinbread instructions to the CONTRIBUTING doc

johnsaigle committed:
Add new phan rule PhanUndeclaredVariableDim to static analysis (#4917)

This adds a rule to make sure that when you access an array that the key you're using makes sense. E.g. if you are accessing an element from an array using a non-numeric key, but the array is actually declared to be a string, this will flag that case.

It fixes all issues identified by phan.

knoll-alex committed:
Delete survey_accounts.md (#5221)

This was a duplicate file of the survey accounts readme.md, stored in the wrong place (help/).

zaliqarosli committed:
Remove scroll bar of frozen headers #5069

Fixes #5068 and second issue in #5067.

This change was already merged before in https://github.com/aces/Loris/pull/3605/files but somehow got reverted.

cmadjar committed:
[DICOM archive - Data Framework] Make DICOM archive menu filter use the DataFrameworkMenu  (#5014)

This updates the DICOM archive module to use the DataFrameworkMenu class.

cmadjar committed:
[raisin bread] Adding EEG entries #5060

This modifies raisin bread to include EEG datasets. In the near future, there will be a PR for a new "Electrophysiology Browser" module and this dataset will be very useful to test that new module!

Note: the /data-raisinbread directory has been updated with the files referenced in the new table entries so everyone in the group should already have access to them on their VM.

zaliqarosli committed:
[jsx/Form.js] Make rendering label for SelectElement and TextboxElement optional (#5208)

This allows the ability to render SelectElements and TextboxElements without its label. If a label prop is provided, the label will render and take up 3 spaces in the row, with the input element taking up the rest of the 9 spaces. This is also the case if the label prop is an empty string. If a label prop is not provided at all, the default is that the label won't be rendered at all and the input element takes up all 12 spaces of the row.

This is useful for modules such as login that still need to be reactified and have input elements without labels.

johnsaigle committed:
Update npm packages (#5220)

Update package-lock.json with results of running `npm audit fix`.

driusan committed:
[Utility] Remove confusing note (#5229)

Remove note which no one seems to understand the origin of
and is causing confusion. Resolves #5227.

ridz1208 committed:
[Statistics] Fix render stats table to only render the MRI error on the MRI page (#5190)

The stats table in 21.0.0 is used in the demographics tab as well as the MRI tab, for the MRI tab, additional check were added to display an error message when the scan type is not associated with a scan_type_scan_done field i nthe MRI parameter form.

The demographics however should never display that error message since the error does not apply to it.

xlecours committed:
[Candidate Parameters] fix permission check (#5118)

The requesting user must have candidate_parameter_edit, not just being authenticated.

Also fixes `\Candidate::singleton((int) $candID);` which was called with a string. It was generating the error:

PHP Fatal error: Uncaught TypeError: Argument 1 passed to Candidate::singleton() must be of the type integer, string given

zaliqarosli committed:
[jsx/Form.js] Add <HeaderElement /> (#5210)

Add HeaderElement as a form element, useful for future instrument rendering in React.

johnsaigle committed:
Add new phan rule PhanTypeSuspiciousStringExpression (#4345)

This rule checks whether value is being converted to a string when it doesn't make sense to do that, such as for booleans.

johnsaigle committed:
Add new phan rule PhanMismatchReturn (#4914)

This adds the phan rule PhanMismatchReturn which flags code that is declared to return a type but in reality returns a different type.

johnsaigle committed:
Correct 'minLength' to 'length' for PSCID generation (#5226)

Fix places that minLength->length was overlooked in #4241.

ridz1208 committed:
[Project] Associate projects to sessions + fix/add foreign keys + c.ProjectID to c.RegistrationProjectID (#4919)

This does 3 related changes to add user projects to LORIS:

1. Add the ProjectID field to sessions in the same manner as centers are affiliated to sessions for future development of project permissions
2. Change the candidate.ProjectID to candidate.RegistrationProjectID to remain consistent with the centerID changes done in #4078
3. Add necessary foreign keys to the SQL and fix existing fields to support the foreign keys from all tables.

maltheism committed:
Allow Directory Structure in Module Namespaces (#5236)

This allows nested hierarchical directory structures to be used in LORIS module namespaces with support for autoloading the classes.

johnsaigle committed:
[Core] Create Password class and refactor password updates (#4411)

This adds a Password class to LORIS in order to improve the type hinting of password strings.

- Valid passwords are created based on strings and must meet or exceed a minimum length and be sufficiently complex as determined by the Zxcvbn library.
- There are no character requirements on passwords; they must be at least 8 characters long and sufficiently complex. This means either mixing many character sets or using longer passwords.
- The Password class checks only whether a Password is valid and well-formed; it does not do any state-based checks or interact with the Database.
- Reduces code duplication across the login and user_accounts modules by leveraging the new functionalities of Utility and Password

johnsaigle committed:
[Phan] Readd PhanTypeSuspiciousStringExpression rule (#5239)

This rule should have been added in #4345 but the phan configuration file update was missed.

AlexandraLivadas committed:
[Test] unittests for NDB_Page.class.inc (#5024)

Add unit tests for the NDB_Page library class.

johnsaigle committed:
[Utility] Deprecate toArray() function (#5240)

This function should've been properly deprecated in LORIS v21 but was missed during the release process.

See #4151.

jesscall committed:
[timepoint_list][bugfix] 'View Candidate Info' button wasn't redirecting (#5233)

Harmonize the "View/Edit Candidate Info" buttons on the timepoint list which went to the same URL and replace with a uniform "Candidate Info".

johnsaigle committed:
[Media] Use DataFrameworkMenu class (#5231)

Update the Media module to use the new DataFrameworkMenu class.

xlecours committed:
Fix php linter (#5249)

With the introduction of php subdirectories in modules, run-php-linter fails. This uses find instead of ls to create the file list.

johnsaigle committed:
[Install] Add -p to mkdir so install doesn't fail (#5245)

Adds -p flag to mkdir so that the install script won't fail when user_uploads/ already exists (for whatever reason) in doc_repo or data_release.

Fixes #5244.

maltheism committed:
[documentation] Updated Mac install Guide (#4319)

Update MacOS install guidelines.

kongtiaowang committed:
[Travis] selenium 3.81 (#5222)

Upgrade from Selenium 2.53 to Selenium 3.81.

Add Headless mode to Travis (saves about 1 minute).

johnsaigle committed:
[Core] Remove unreachable code in LORIS using phan (#4967)

Adds a phan plugin to create a rule to detect unreachable code and remove unreachable code.

johnsaigle committed:
[Core] Use CandID class in LORIS (#4350)

Use the CandID class for built-in verification throughout LORIS. Also cleans up code that was made obsolete by #4241.

Resolves #4295.

knoll-alex committed:
Created a readme file for the QC module

jesscall committed:
[media] Use study's start/end date as min/max values for Date of Administration field (#5248)

When uploading a form to media module, you cannot currently enter a date greater than 2017-12-31. The date of administration field is currently hard-coded to have a minYear of '2000' and a maxYear of '2017'.

Min and max year are now determined by the study's start and end year (taken from configuration).

johnsaigle committed:
Add syntax highlighting in raisinbread README (#5257)

Add language to code blocks so that the GitHub frontend renders with syntax highlighting.

johnsaigle committed:
Instruct new developers not to create new branches on the repo (#5254)

Update CONTRIBUTING.md to tell developers to work based off of forks instead of committing to the repo.

Resolves #5209.

ridz1208 committed:
[Core] Fix static PSCID generation (#5281)

[Core] Fix static PSCID generation

zaliqarosli committed:
Update package.json / node_modules (#4968)

Jkat committed:
Small fix in README (#5276)

Fix typo in README.

johnsaigle committed:
Add PHP 7.4 (snapshot build) to TravisCI tests (#5265)

Add PHP 7.4 as an allowed failure to Travis CI.

johnsaigle committed:
Update php-ast in Travis (#5266)

Update php-ast to 1.0.1

maltheism committed:
[tools/install.sh] fix for other operating systems (#5295)

Allows the install script to continue if the command "hostnamectl" is not found.

Jkat committed:
Fixing Acknowledgements TestPlan to Markdown syntax (#5277)

Fixing TestPlan to use a GitHub flavored Markdown supported header syntax.

ridz1208 committed:
[Raisinbread] update migration log major (#5284)

johnsaigle committed:
[Utility] Add isPositiveInteger function (+ unit tests) (#4425)

When checking data coming from e.g. the database or a form, there are some weird subtleties with using intval(). It'll return 0 on an input of null which makes it hard to differentiate between inputs that are null and 0 and those that are unset. It also has non-intuitive behaviour for array and object inputs, see: https://secure.php.net/manual/en/function.intval.php

This PR adds a simple function to check whether a given value is a positive integer as well as unit tests to prove that it works.

johnsaigle committed:
[Login] Fix static calls to User class to be non-static (#5247)

The password expiry code was making static calls to updatePassword and passwordChanged in the User class which are non-static methods. This created errors such as PHP Fatal error: Uncaught Error: Using $this when not in object context, making it impossible to update an expired password due to an oversight in #4411.

zaliqarosli committed:
[BVL Feedback] Add dynamictable and fix margin-top (#5089)

Add carousel to 'Open Thread Summary' table and horizontal scrolling to 'Feedback Threads' table.

Fixes #5074.

ridz1208 committed:
[Raisinbread] update migration log (#5283)

Update log of which patches are applied to RaisinBread on minor.

johnsaigle committed:
[Core] Add '200 OK' and '201 Created' responses to src/ (#4949)

Add OK and Created response classes to aid creation of JSON responses.

ridz1208 committed:
Merge tag 'v21.0.2' into PUSH_v21.0.2_INTO_MINOR

ridz1208 committed:
Merge branch 'PUSH_v21.0.2_INTO_MINOR' into PUSH_v21.0.2_INTO_MAJOR

driusan committed:
Merge pull request #5302 from ridz1208/PUSH_v21.0.2_INTO_MINOR

PUSH OF v21.0.2 INTO MINOR

driusan committed:
Move Response objects to explicit JSON namespace (#5301)

Move ambiguously named HTTP response classes which only handle JSON responses to an explicit JSON sub-directory/namespace.

This should avoid confusion by making the fact that they're only useful for JSON more explicit.

ridz1208 committed:
user unit test

racostas committed:
[Statistics / Demographic Statistics] Fix permission issue where users were allowed to see statistics from sites they don't have access to. (#5315)

A user with role Data Entry was able to see demographic statistics from sites it does not belong/have access

For example if there are two users and two sites:
user Site 1 is associate only to Site_1.
user Site 2 is associate only to Site_2.

(Both of them have the role Data Entry and nothing more in the permissions)
When going to MainMenu->Reports->Statistics->Demographic Statistics
The user of Site 1 is able to see the statistics for Site 2 and vice versa

Now each user should be able to see only the demographic statistics for the sites it belong/have access

Fixes #5086

driusan committed:
[Data Tables] Improve efficiency of DataTable filtering for large data tables (#5299)

This improves the efficiency of the DataTable JSX component when dealing with
large data tables. When counting the number of rows which match the filters
for the count in the header/footer, it returns an array of indexes which matched
the filter rather than returning a count. This prevents the need to needlessly
duplicate the work of tracking which rows match for the data table.

The indexed array is then sorted, and we work with that sorted list of indexes
instead of repeatedly going over this.props.data after the first time the matching
rows are calculated.

johnsaigle committed:
[Configuration] Path settings must be valid paths (#4023)

Ensure that path corresponds to a real path on the server when a config setting is set in LORIS.

johnsaigle committed:
Re-add phan rules that were accidentally deleted

xlecours committed:
[Api] /candidates and sub-endpoints (#4244)

Finish conversion of API from ad-hoc PHP scripts to proper LORIS module.

jesscall committed:
[candidate_parameters] Fixed breadcrumb (#5313)

When clicking on the breadcrumb in Candidate Parameters, the CandID was not being included in the link and thus was throwing a 500 error (specifically in the Consent Status tab).

xlecours committed:
Fix Namespace of NotModified response for ETag middleware (#5316)

Add JSON portion to class namespace reference.

johnsaigle committed:
Fix phan errors

driusan committed:
Merge pull request #5304 from ridz1208/PUSH_v21.0.2_INTO_MAJOR

PUSH OF v21.0.2 INTO MAJOR

kongtiaowang committed:
[install README CentOS and Ubuntu] 21 install updates, including make - bugfix (#5203)

Add this [Run make (for production instances) or make dev (for development sandboxes). ] into readme

to help solve issue #5197
related to #5198
Resolves #3946

xlecours committed:
[API] Make candidate images permission "protected" (#5326)

- Moving Phantoms and Scans filters in src
- Adding filters to the timepoint images provisioner

- force add modules/api/php/endpoints/project folder to git tracked files
- force add modules/api/php/views/project folder to git tracked files

fixes #5341

johnsaigle committed:
Add array return type to pselect (#5317)

Database->pselect() now always returns an array, thus removing the need to check if its result is an array before proceeding.

This should be safe as it is just a wrapper for the execute() function which already returns an array.

johnsaigle committed:
[Install] Use "command" instead of "which" to determine if a binary is installed (#5322)

Use `command -v` which is POSIX standard. `which` can also give weird path results sometimes so we should avoid it.

johnsaigle committed:
Update phpmd (#5350)

Update phpmd to latest version.

ridz1208 committed:
2019 10 09 wiki to minor (#5305)

Bringing the new wiki to the minor branched to be progressively developed in beta mode until it completely replaces the WIKI.

johnsaigle committed:
Update phan to latest minor version (#5333)

johnsaigle committed:
[DQT] Fix 500 error (#5345)

"ConfigException" doesn't exist and is causing LORIS to crash (for the wrong reason) when an exception is thrown.

johnsaigle committed:
ReadTheDocs initialization (#5242)

Initialize ReadTheDocs configuration so that it can be tested as a replacement for the GitHub wiki.

johnsaigle committed:
[ReadTheDocs] Change nav to pages to prevent build errors (#5371)

Running into a known issue when ReadTheDocs and Mkdocs compatibility: readthedocs/readthedocs.org#4922

Use fix suggested in above issue.

jesscall committed:
[data_team_helper] fixed broken link to visit (#5378)

Fixes #5376.

racostas committed:
[Statistics / Imaging Statistics] Fix permission issue where users were allowed to see imaging statistics from sites they don't have access to. (#5339)

A user with role Data Entry was able to see imaging statistics from sites it does not belong/have access

For example if there are two users and two sites:
user Site 1 is associate only to Site_1.
user Site 2 is associate only to Site_2.

(Both of them have the role Data Entry and nothing more in the permissions)

When going to MainMenu->Reports->Statistics->Imaging Statistics

The user of Site 1 is able to see the imaging statistics for Site 2 and vice versa.

maltheism committed:
[publication] 21.0 release publication email verify fix (#4730)

Incorporated the frontend browser check for validation of email by changing type to email.

jesscall committed:
[media] Use study's start/end date as min/max values for Date of Administration field #5367

Date of administartion min and max year are currently hard-coded, but should use the config values of study start and end date.

Resolves #5361.

kongtiaowang committed:
[user_account] can't add or edit a user on major (#5370)

Fix fatal error when trying to add or edit user.

Resolves #5368.

cmadjar committed:
[imaging_uploader] Removed unused variables from imaging_uploader.class.inc (#5307)

Clean up the imaging_uploader.class.inc file by removing all declaration of variables that are never used.

johnsaigle committed:
[Media] Rename date_uploaded column to last_modified (#5256)

The Date Uploaded column is not named accurately. It is a timestamp value that updates to the current timestamp upon edit.

Given this behaviour, rename it to "Last Modified".

driusan committed:
[Script] Add script to generate fake candidates (#5300)

This adds a script used to generate a large number of candidates
for #5299. The candidates are not realistic or even necessarily
valid, but only intended to see how the candidate_list data table
holds up with a large number of candidates (I tested with up to
20,000.)

This needs a lot of work to be made robust enough to be useable
for general purpose testing and is unlikely to be maintained, so
is added directly to the "deprecated" directory.

maltheism committed:
[jsx/Form.js] new slider component!  (#5280)

Add SliderElement JSX element which can be used by instruments in the new instrument builder.

cmadjar committed:
[imaging uploader] Automatically populate PSCID, CandID & visit label based on the filename upon upload (#5306)

Modified the behaviour of the upload tab of the imaging uploader module. PSCID, CandID and Visit Label fields are now automatically populated based on the name of the selected file to be uploaded.

The DICOM header and file name still must match as a form of double data entry/verification.

johnsaigle committed:
Fix edge-case that resulted in 500 error (#5320)

There was a slight issue with the logic during the install process that swallowed a DB exeption in favour of returning false; however, the return value of this function was never checked so the code kept trying to function without a DB connection. This could give a 500 error in some cases, making the install look messy and hard to debug.

HenriRabalais committed:
Fixing bugs introduced when optimizing DataTable (#5381)

When optimising the Datatable, #5299 introduced some subtle bugs in the file. This fixes some of those 

- It maps the headers to values to allow the this.props.getFormattedCell to function properly.
- It pushes rowData to filterData so that when downloading the CSV, only the filtered data is passed.
- Checks for cases where data=null because null.toLowerCase() cannot be executed.

There are still issues with pagination, which remain to be fixed.

johnsaigle committed:
[DQT] Fix PHP Notices (#5347)

When ajax/datadictionary.php is called, Notices are generated when the category or key params are checked when not present in the request. Add isset check to avoid them.

kongtiaowang committed:
[install.sh] Cleanup for 21: dependency checks, logic, user messages (#5205)

Clean up install script and instruct user's to run make [dev] after.

johnsaigle committed:
[Cand. Params] Users with cand param edit permission should be able to view module (#5373)

Previously only users with candidate_parameter_view could view the module. This allows users with candidate_parameter_edit to view it also.

cmadjar committed:
[brainbrowser] add possibility to read nii.gz files with BB (#5314)

nii.gz files could not be opened with the brainbrowser module. This modifies the way we read the image so that if the file is gzipped, it is decompressed first before reading the content of the file.

HenriRabalais committed:
[Database] Update database class (#5290)

- Created a check to make sure that both the ignore and onDuplicateInsert flags are not used simultaneously in the _realInsert function.
- Removed the unsetting of the unique key when using the pselectWithIndexKey function for situations in which the value is necessary in the results tuple.

johnsaigle committed:
[PHPCS] Update to latest version (#5349)

johnsaigle committed:
[ReadTheDocs] Fix build failure by adding explicit dependencies (#5385)

Old versions of mkdocs (used by ReadTheDocs) have incompatibilities with Python 3.7. readthedocs/readthedocs.org#5332

This adds explicit settings for our project so that ReadTheDocs will run with Python 3.7 on a relatively new version of mkdocs. This should fix our current build issue.

cmadjar committed:
[Imaging Browser - Data Framework] Make Imaging Browser menu filter use the DataFrameworkMenu (#5022)

This updates the imaging browser module to use the DataFrameworkMenu class. This should simplify the maintenance of the class and make it easier to use project permissions when implemented.

ridz1208 committed:
[User accounts] Add project option to user accounts (#5009)

Add the functionality to affiliate users to specific projects.

kongtiaowang committed:
[new_profile] can't create a new profile on major (#5329)

Fix type error preventing creation of new profiles.

driusan committed:
Merge remote-tracking branch 'aces/minor' into HEAD

driusan committed:
Fix PHPCS errors

driusan committed:
Fix phan errors except for ones related to API

driusan committed:
Delete invalid SQL statement

driusan committed:
Fix column name in surveyaccounts test

driusan committed:
Fix number of candidates in media test

driusan committed:
Move population of project table to before first reference in setup function

ridz1208 committed:
Update survey_accountsTest.php

driusan committed:
PHPCBF

ridz1208 committed:
Update survey_accountsTest.php

ridz1208 committed:
Update survey_accountsTest.php

ridz1208 committed:
[Core] Create UserProjectMatch filter (#5398)

Creation of the userProjectMatch filter for the data framework and implementing it on the candidate_list and media modules.

With this filter you should now be able to restrict the data in the data-table by both sites and projects.

driusan committed:
Merge remote-tracking branch 'aces/major' into MergeMinorMajor

driusan committed:
Use Project filter in imaging browser

xlecours committed:
Fix api (#47)

Fix API after branch merge.

driusan committed:
PHPCBF

driusan committed:
fix column name in media module

driusan committed:
Merge pull request #5425 from driusan/MergeMinorMajor

Push minor branch into major

ridz1208 committed:
Add project to PSCID generation  (#5285)

Added option to generate PSCIDs using the project alias instead of the site alias.

cmadjar committed:
added 40 fake entries to the acknowledgements module (#5435)

Add 40 fictitious acknowledgements to raisin bread so the acknowledgement module can be tested properly.

cmadjar committed:
update to the acknowledgement module test plan (#5405)

This updates the test plan of the acknowledgements module.

ridz1208 committed:
[SQL] fix unsourcable schema (#5465)

Missing alias in default project was preventing schema sourcing.

cmadjar committed:
update to the test plan of timepoint list (#5442)

Small updates to the test plan for the timepoint list module.

cmadjar committed:
[imaging_browser] fixed query for the imaging browser (#5444)

This fixes the query for the imaging browser by taking the ProjectID from the session table instead of the candidate table.

ridz1208 committed:
[Brainbrowser] Test plan update (#5475)

Clarify brainbrowser test plan.

ridz1208 committed:
[Project] Fix instances of c.project coming from minor (#5445)

Fix instances of c.ProjectID coming from the merge of minor into major, where the column was renamed.

ridz1208 committed:
Merge tag 'v21.0.3' into PUSH_v21.0.3_INTO_22.0

ridz1208 committed:
phpcs

ridz1208 committed:
missing namespace

driusan committed:
Merge pull request #5498 from ridz1208/PUSH_v21.0.3_INTO_22.0

Push v21.0.3 into 22.0

jesscall committed:
[candidate_parameters][bugfix] getData returning candID object was breaking react (#5518)

Fix bug where forms were loading and then disappearing quickly afterwards on the candidate_parameters page.

When returning the CandID object in the getData.php file, incorrect data was returned because React was expecting a string and cannot render the object that was passed.

Resolves #5434

cmadjar committed:
[imaging_browser] Fix imaging browser query merge mix ups... (#5528)

Fix query in imaging browser that was broken by schema incompatibility on a branch merge.

cmadjar committed:
[acknowledgement] Remove edit buttons from UI when the user has no edit permission #5428

This changes allows the display of the 'Add Acknowledgement' button only if the user has acknowledgements_edit permission.

zaliqarosli committed:
[Document Repository] Fix duplicate file error throwing (#5531)

Fixes #5453

cmadjar committed:
fix UI for statistics when on mobile mode (#5541)

Fixes minor UI issue in dropdown in <900px wide screens.

Resolves #5460.

sruthymathew123 committed:
[Statistics] Fix Behavioural Statistics Value Not Updating Issue #5535 Open

Fix misnamed column name for RegistrationProjectId resulting from branch merge.

Fixes #5463.

johnsaigle committed:
[Configuration] Fix issue where a Project could not be added without subproject data (#5529)

Projects can now be added without requiring that subproject data be configured at the same time. This error was encountered during v.22 release testing.

Resolves #5511.

xlecours committed:
[Imaging Uploader] Fix error where uploaded image did not display on front-end #5513

Fixes #5413.

johnsaigle committed:
[User Accounts] Fix password bugs: "confirm password" must match; password cannot be email (#5426)

Password and Confirm Password on the front-end don't need to match on major. This check was removed by mistake during Password refactoring.

Fixes #5416, #5417

zaliqarosli committed:
[DocRepo] Update test plan for 22.0-release (#5525)

Remove test for functionality that was never implemented and clarify other points.

Fixes #5456.

driusan committed:
[Data Team Helper] Remove untrue statement from DTH README (#5521)

Remove reference to the Data Team Helper module being listed as
"Quality Control" in the LORIS menu, (which in fact links to the
quality_control module.)

Fixes #5503.

ycAbout committed:
[Issue Tracker] update test plan to delete items not relevant (#5540)

Remove reference to function which was deleted in test plan.

maltheism committed:
[installdb.php] Fixes sourceSchema not reporting error (#5497)

Fix error reporting when something prevents the schema from being sourced properly.

ridz1208 committed:
[configuration] clean-up of testplan for readability (#5512)

Most changes are whitespace, added new lines for readability in raw format.

johnsaigle committed:
Update README for 22 release (#5523)

cmadjar committed:
[brainbrowser] Add permissions to view the module (#5530)

Add same permissions to brainbrowser as imaging browser, as described in the test plan.

johnsaigle committed:
Fix broken link in CONTRIBUTING.md (#5549)

Fix broken link in CONTRIBUTING.md.

Fixes #5547.

zaliqarosli committed:
[Notification/Permissions] Fix raisinbread data rels (#5537)

Fixes data that fixes #5457.

Jkat committed:
[Instrument Manager] Improving Instrument Manager TestPlan (#5519)

Fix small mistakes in instrument manager test plan.

ycAbout committed:
Improve instrument builder test plan (#5534)

Remove references to non-existent features; improve readability.

ridz1208 committed:
[data_release] Add new permission for viewing only (#5557)

This adds a new permission to the database to allow users view/download access to the data release module. the other 2 permissions that already exist give the user slightly more power, this one is the most basic access.

Addresses #5496

ridz1208 committed:
[create_timepoint] Fix null project bug (#5553)

Fixed bug #5418 and changed behaviour to only display projects user is affiliated with.

ridz1208 committed:
[datadict] Test plan improved readability and fixed script location (#5558)

Fix path for script location in test plan and fix whitespace for readability.

cmadjar committed:
[imaging_uploader] fix forbidden file types that could be uploaded (#5536)

This fixes the issue of any file type being able to be uploaded when the 'Phantom Scans' was set to 'Yes'.

See #5526.

cmadjar committed:
[dicom archive] fixed the metadata not being displayed in the view details page (#5450)

The metadata displayed at the top of the DICOM archive view details page were not populated. This fixes the bug.

Fixes #5447

ridz1208 committed:
[Create timepoint] testplan update (#5568)

- Clearer enumeration of possible test cases after selecting subproject
- Added project test cases

xlecours committed:
[bvl feedback] Fix thread status update (#5522)

It fixes the status update and adds various error message and HTTP status codes.
While the error code are still not handled by the React app, the update now works.

Fixes #5467.

zaliqarosli committed:
[document_repository] clean up directory content and let filter without folders (#5572)

Remove extraneous files that actually belong to subdirectories while adding a checkbox to do global filtering of files without the directory structure, so that documents can still be searched for.

Fixes #5449, #4857

johnsaigle committed:
[New Profile] Fix TypeError when creating candidate with "user" generation method" (#5555)

Fix type error when creating a new profile.

sruthymathew123 committed:
[Examiner] Fixing Errors while saving the form (#5545)

Fix issues #5423 and #5544.

sruthymathew123 committed:
[Examiner] Removing Training Module Reference from Examiner Readme (#5579)

Remove reference to training module which was removed from LORIS.

cmadjar committed:
fix not being able to see the acknowledgement page when have edit permission only (#5584)

Make edit permission imply view permission for acknowledgement module.

johnsaigle committed:
[MRI_Violations] Fixes permission bug; integration test to confirm bug fix; test plan update (#5546)

Allows users with the 'edit' permission to view the module even if they don't have 'view' permissions.

Adds an integration test to the above + code cleanup.

Rewrites the test plan to be up-to-date and easier to follow.

    Resolves #5419
    Resolves #5421
    Resolves #5422

ridz1208 committed:
[Configuration] Added new Alias field to the front end (#5556)

Aliases were added to the Project table, but the frontend was not updated to reflect that. Add "Alias" to the frontend.

Fixes #5539

johnsaigle committed:
[Testing] Add unit tests for User::passwordChanged (#5514)

Adds unit tests to the passwordChanged function.

ridz1208 committed:
[Configuration] display radio buttons when boolean option does not have a default option (#5592)

This fixes the rendering of boolean fields in the configuration module. without this change a boolean field without a default option set (field defined in ConfigSettings but no entry in Config table) renders as a text entry in the configuration module. it should render as true/false radio buttons with non selected with this change.

HenriRabalais committed:
[Media] Added Project to Columns; Added Language to Edit File Form (#5585)

This fixes 2 issues:

#5581: The MediaFileProvisioner was adding the project column and filter, but the edit window had not been updated resulting in it not loading.

#5459: There was no field for 'Language' in the edit media form, and therefore there was no way to edit language. Language has now been added as a field and is editable.

maltheism committed:
[install CentOS] Fixes httpd apache module for php is missing (#5492)

Enable PHP yum repository in CentOS instructions.

Fixes #5383.

maltheism committed:
[install CentOS] mariadb packages to lowercase correction (#5491)

Use modern version of MariaDB on CentOS.

ridz1208 committed:
[brainbrowser] testplan update (#5598)

Update BrainBrowser test plan for 22.0 release.

johnsaigle committed:
[Test] Restore test_all_fields instrument for use in testing Instrument Manager (#5594)

Restore test_all_fields.linst file needed for instrument test plan.

Fixes #5527

johnsaigle committed:
[User] Fix TypeError in password verification (#5602)

Fixes TypeError introduced by #5426.

Resolves #5601

ridz1208 committed:
[configurations] Allow for both 0/1 or true/false to be used (#5605)

This restores a behaviour from #5592 while keeping the bugfix.

Previous behaviour was intentionally maintaining 1/0 values when previously saved value was 1/0 and true/false values when previously saved value was true/false. That is due to a lack of standard when coding with boolean values within modules.

The PR linked above converted all values to 1/0 automatically which is problematic because some code sections will be expecting a true or false to be returned from the getSetting('something') call.

cmadjar committed:
[imaging_browser] Fix permissions of the view session page (#5589)

Users with phantom permissions could still view any view session page if given a link to that page. This fixes the permissions of the view session page of the imaging browser to be consistent with the viewsession page and the menu page.

ridz1208 committed:
[RB] Add a default for the config setting instrumentResetting (#5593)

HenriRabalais committed:
[DataTable] Resolved Pagination and CSVDownload Issues (#5600)

This resolves the following issues:
#5566: The pagination is now functional on the DataTable.
#5458: All filtered rows will be downloaded when clicking on 'Download CSV' button.

johnsaigle committed:
[User Accounts] Update password strength rules documentation (#5614)

Update user accounts module to properly display the new password strength guidelines. These match the ones on the password expiry form in the login module.

ridz1208 committed:
Rename Quality control and data team helper to imaging/behavioural QC and reorganise menu (#5620)

Refactor the modules used for quality control to be more appropriately named. There is now an "imaging quality control" and "behavioural quality control" module which are independent, since they target different audiences.

christinerogers commented:
Can you rebase and issue this as a PR aces/22.0-release today?  
Also when you do so --  it's important to: 
- [ ] tag Leigh MacIntyre, her github handle is: `leighmac` 

And either note very visibly / make the following upcoming updates: 
- [ ] Per recent changes in [Rida's PR 5620](https://github.com/aces/Loris/pull/5620 ) : Bvl tab will be gone, i.e. ti's just the Imaging QC module now. Match the exact language of the PR: https://github.com/aces/Loris/pull/5620.

knoll-alex committed:
updates based on PR 5620

changed "data team helper" --> "behavioral quality control"
changed "quality control module" --> "imaging quality control module"
removed mention of behavioral tab

christinerogers committed:
rewrote a lot

* change of focus more to Imaging 
* remove BVL elements like DDE
* Important clarification:  you can't do imaging QC in this module

christinerogers committed:
input from Rida 

also adding a little more text. maybe redundant?

knoll-alex committed:
Rename readme.md to README.md

knoll-alex committed:
fixed spacing

christinerogers committed:
[skip travis] based on Dave's change request

knoll-alex committed:
updated file name to live in imaging_qc folder

christinerogers committed:
[skip ci] correct ImgUpl info

dave was right.

christinerogers committed:
fix and smoothing [skip ci]

fixed a old typo `reviwed`
smoothed the wording
re-checked

christinerogers committed:
correcting header level for modules [skip ci]

christinerogers committed:
caveat module broken without MRI PF [skip ci]

christinerogers committed:
typo fix and better wording [skip ci]

christinerogers committed:
fix language. [skip ci]

knoll-alex committed:
replaced subheadings for modules with bullets

knoll-alex committed:
fixed spacing

ridz1208 committed:
Update README.md

christinerogers committed:
update w Rida and Dave's requests [skip ci]
